extensions/transfer/transfer-provision-aws/src/main/java/com/microsoft/dagx/transfer/provision/aws/s3/S3BucketProvisionedResource.java

```diff
@@ -28,9 +36,15 @@
     @JsonProperty
     private String bucketName;
 
+    private S3CompletionChecker checker;
+
     private S3BucketProvisionedResource() {
     }
 
+    private S3BucketProvisionedResource(ClientProvider clientProvider, RetryPolicy<Object> retryPolicy) {
```

Comment:
We shouldn't mix completion checking with the provisioned resource. 

A provisioned resource is a representation that can be serialized and moved about in the system or passed to other trusted systems. 

In my view, completion checking is similar to a provisioner in that it is responsible for performing a function with specialized knowledge of how data is stored and when it is complete. I would tend to model this as a separate functional interface (as opposed to Supplier) so that people can quickly find implementation types in their favorite IDE.